presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkRowBuffer.java

```diff
@@ -66,48 +62,21 @@ public void setNoMoreRows()
         }
     }
 
-    public List<PrestoSparkMutableRow> get()
+    public PrestoSparkRowBatch get()
             throws InterruptedException
     {
-        BufferedRows bufferedRows = null;
+        PrestoSparkRowBatch rowBatch = null;
         synchronized (monitor) {
             while (buffer.isEmpty() && !finished) {
                 monitor.wait();
             }
             if (!buffer.isEmpty()) {
-                bufferedRows = buffer.poll();
+                rowBatch = buffer.poll();
             }
-            if (bufferedRows != null) {
-                memoryManager.updateMemoryUsage(-bufferedRows.getRetainedSizeInBytes());
+            if (rowBatch != null) {
+                memoryManager.updateMemoryUsage(-rowBatch.getRetainedSizeInBytes());
```

Comment:
This is unrelated to this PR: but maybe add some comments explain it is intended that we reduce the memory accounting eagerly when the `rowBatch` is polled.  But at most one single row batch will be under counted with about ~1MB so it's a good enough approximation. (as discussed in https://github.com/prestodb/presto/pull/14522#discussion_r432280758)

Reply:
Added a comment